haystack/components/connectors/openapi_service.py

```diff
@@ -0,0 +1,145 @@
+    are then used to invoke the method on the OpenAPI service. The response from the service is returned as a
+    ChatMessage.
+
+    Before using this component, one needs to register functions from the OpenAPI specification with LLM.
+    This can be done using the OpenAPIServiceToFunctions component.
+    """
+
+    def __init__(self, service_auths: Optional[Dict[str, Any]] = None):
+        """
+        Initializes the OpenAPIServiceConnector instance
+        :param service_auths: A dictionary containing the service name and token to be used for authentication.
+        """
+        openapi_imports.check()
+        self.service_authentications = service_auths or {}
+
+    @component.output_types(service_response=Dict[str, Any])
+    def run(self, messages: List[ChatMessage], service_openapi_spec: Any):
+        """
+        Processes a list of chat messages to invoke a method on an OpenAPI service. It parses the last message in the
+        list, expecting it to contain an OpenAI function calling descriptor (name & parameters) in JSON format.
+
+        :param messages: A list of ``ChatMessage`` objects representing the chat history.
+        :type messages: List[ChatMessage]
+        :param service_openapi_spec: The OpenAPI JSON specification object of the service.
+        :type service_openapi_spec: JSON object
+        :return: A dictionary with a key ``"service_response"``, containing the response from the OpenAPI service.
+        :rtype: Dict[str, Any]
+        :raises ValueError: If the last message is not from the assistant or if it does not contain the correct payload
+        to invoke a method on the service.
+        """
+
+        last_message = messages[-1]
+        if not last_message.is_from(ChatRole.ASSISTANT):
+            raise ValueError(f"{last_message} is not from the assistant.")
+
+        method_invocation_descriptor = self._parse_message(last_message.content)
+
+        # instantiate the OpenAPI service for the given specification
+        openapi_service = OpenAPI(service_openapi_spec)
+        self._authenticate_service(openapi_service)
+
+        service_response = self._invoke_method(openapi_service, method_invocation_descriptor)
+        return {"service_response": [ChatMessage.from_user(str(service_response))]}
+
+    def _parse_message(self, content: str) -> Dict[str, Any]:
+        """
+        Parses the message content to extract the method invocation descriptor.
+
+        :param content: The JSON string content of the message.
+        :type content: str
+        :return: A dictionary with method name and arguments.
+        :rtype: Dict[str, Any]
+        :raises ValueError: If the content is not valid JSON or lacks required fields.
+        """
+        if not self._is_valid_json(content):
+            raise ValueError("Invalid JSON content, cannot parse invocation message.", content)
```

Comment:
I guess we can do this directly in `_parse_message()`, we can avoid parsing JSON two times and also we can show a better error message.